### PR TITLE
Add General NATS Architecture

### PR DESCRIPTION
**Pull request by ColinSullivan1**
For convenience, the link to view the document rendered is [here](https://github.com/nats-io/roadmap/blob/add-arch/architecture/ARCHITECTURE.md).

### CODE REVIEW

File: architecture/ARCHITECTURE.md [hunk 1]
```diff
@@ -1,36 +1,36 @@
...
+Running a single NATS server introduces a SPOF.  In order to provide high availability and scalability, NATS servers support full mesh clustering. Each server is connected to all other servers in the cluster.  There is a one-hop message routing maximum, ensuring messages will never loop througout a cluster.  The servers communicate with each other using a [server-to-server clustering protocol](http://nats.io/documentation/internals/nats-server-protocol/) over a TCP connection.  The protocol supports "discovery" to propogate topology information and changes in real-time with other members of the cluster and clients.  Thus, servers can be dynamially added or removed from a cluster at runtime with zero downtime.  Ideally, a client will have at least 2 addresses of "seed" servers.
```
Comment (kozlovic) [inline architecture/ARCHITECTURE.md]: Typo `dynamially` -> `dynamically`.

File: architecture/ARCHITECTURE.md [hunk 2]
```diff
@@ -1,36 +1,36 @@
...
+NATS messaging is comprised of core NATS, and NATS streaming.  Core NATS supports at-most-once delivery, is designed to be lightweight, performant, and always available.  NATS Streaming supports log based persistence providing at-least-once delivery, replay of messages, and subscription continuity (durable subscribers).
 
 # Core NATS
 
-Core NATS is a broker based client-server [messaging](http://nats.io/documentation/concepts/nats-messaging/) software supporting the [publish/subscribe](http://nats.io/documentation/concepts/nats-pub-sub/), [request/reply](http://nats.io/documentation/concepts/nats-req-rep/), and [work queue](http://nats.io/documentation/concepts/nats-queueing/) messaging patterns.
+Core NATS is the ideal [messaging](http://nats.io/documentation/concepts/nats-messaging/) software for [publish/subscribe](http://nats.io/documentation/concepts/nats-pub-sub/), [request/reply](http://nats.io/documentation/concepts/nats-req-rep/), and [work queue](http://nats.io/documentation/concepts/nats-queueing/) messaging patterns.
```
Comment (ColinSullivan1) [inline architecture/ARCHITECTURE.md]: IMO `a broker based client-server` is more neutral and technically oriented...

File: architecture/ARCHITECTURE.md [hunk 3]
```diff
@@ -63,11 +63,11 @@ The typical flow of a NATS client is very straightforward:
 
 # Streaming server
 
-The [NATS streaming server](https://github.com/nats-io/nats-streaming-server) and streaming clients are a different that core NATS.  Conceptually it's useful to consider NATS Streaming as a layer above NATS - streaming servers are actually core NATS **clients**.  This offers flexibility in allowing NATS streaming servers to have dedicated hosts, distributing work.
+The [NATS streaming server](https://github.com/nats-io/nats-streaming-server) and streaming clients are a different protocol than core NATS.  Conceptually it's useful to consider NATS Streaming as a layer above NATS - streaming servers are actually core NATS **clients**.  This offers flexibility in allowing NATS streaming servers to have dedicated hosts, distributing work.
```
Comment (ColinSullivan1) [inline architecture/ARCHITECTURE.md]: ...and streaming clients use a different...

**Comment by kozlovic** [comment]
@petemiron LGTM.

**Pull request merged**